File: go/1-1-convert-hex-to-base64.go
package main

import (
	"encoding/base64"
	"encoding/hex"
	"fmt"
	"log"
)

func main() {
	src := []byte("49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d")

	rawBytes := make([]byte, hex.DecodedLen(len(src)))
	n, err := hex.Decode(rawBytes, src)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(rawBytes)
	// [73 39 109 32 107 105 108 108 105 110 103 32 121 111 117 114 32 98 114 97 105 110 32 108 105 107 101 32 97 32 112 111 105 115 111 110 111 117 115 32 109 117 115 104 114 111 111 109]

	fmt.Println(n)
	// 48

	fmt.Printf("%s\n", rawBytes[:n])
	// I'm killing your brain like a poisonous mushroom

	str := base64.StdEncoding.EncodeToString(rawBytes)

	fmt.Println(str) // SSdtIGtpbGxpbmcgeW91ciBicmFpbiBsaWtlIGEgcG9pc29ub3VzIG11c2hyb29t
}

File: node.js/1-1-convert-hex-to-base64.js
var src = new Buffer('49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d', 'hex')

console.log(Buffer.from(src, 'hex'))
// <Buffer 49 27 6d 20 6b 69 6c 6c 69 6e 67 20 79 6f 75 72 20 62 72 61 69 6e 20 6c 69 6b 65 20 61 20 70 6f 69 73 6f 6e 6f 75 73 20 6d 75 73 68 72 6f 6f 6d>

console.log(src.toString())
// I'm killing your brain like a poisonous mushroom

console.log(Buffer.from(src, 'hex').toString('base64'))
// SSdtIGtpbGxpbmcgeW91ciBicmFpbiBsaWtlIGEgcG9pc29ub3VzIG11c2hyb29t
